FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m

```diff
@@ -62,6 +62,21 @@ - (void)populateNotificationContent:(UNMutableNotificationContent *)content
 }
 
 #if TARGET_OS_IOS || TARGET_OS_OSX || TARGET_OS_WATCH
+- (NSString *)fileExtensionForResponse:(NSURLResponse *)response {
+  NSString *const kDefaultExtension = @".png";
+  NSString *const kImagePathPrefix = @"image/";
+  NSString *suggestedPathExtension = [response.suggestedFilename pathExtension];
+  if (suggestedPathExtension != nil && suggestedPathExtension.length > 0) {
+    return [NSString stringWithFormat:@".%@", suggestedPathExtension];
+  }
+  if (response.MIMEType != nil && [response.MIMEType containsString:kImagePathPrefix]) {
+    NSString *mimeTypePathExtension =
+        [response.MIMEType stringByReplacingOccurrencesOfString:kImagePathPrefix withString:@""];
+    return [NSString stringWithFormat:@".%@", mimeTypePathExtension];
+  }
+  return kDefaultExtension;
```

Comment:
I don't know image converting enough to know that if it's okay to return a default format like "png" if it is not a png image?
maybe if neither format is valid, we should return empty string and let it fail with the error before?
I don't know if it's also secure to download file that might not be image at all. wdyt?

Reply:
I figured we'd want to set to ".png" in case the metadata wasn't stored, but agreed we can let it error instead.